tracks/go/exercises/pig-latin/mentoring.md

```diff
@@ -0,0 +1,72 @@
+# Pig Latin
+
+### Reasonable solutions:
+
+There is a (somewhat) reasonable solution using regular expressions, however most students are likely to use a combination of `strings` funtions or string slicing to solve this problem.
+
+Solutions vary considerably, but most reasonable solutions should demonstrate these three insights:
+
+1. Decoupled logic to break down a sentence into words and reform the Pig Latin sentence.
+
+```
+func Sentence(sentence string) string {
+	words := strings.Fields(sentence)
+	pigWords := make([]string, 0, len(words))
+	for _, w := range words {
+		pigWords = append(pigWords, pigWord(w))
+	}
+	return strings.Join(pigWords, " ")
+}
+```
+
+2. Identify that some number of consonants (according to the rules) will be added moved to the end of the word and "ay" will be added to the end of the word.
+
+```
+func pigWord(word string) string {
+	consonants := consonantCount(word)
+	return fmt.Sprintf("%s%say", word[consonants:], word[:consonants])
+}
+```
+
+3. Identify how many letters count as consonants without overly convoluted logic.
+
+```
+func consonantCount(word string) int {
+	if strings.HasPrefix(word, "xr") || strings.HasPrefix(word, "yt") {
+		return 0
+	}
+	for i, r := range word {
```

Comment:
More readable var names might be good here.

Reply:
`i` is a [well-accepted name for an index variable](https://go.dev/talks/2014/names.slide#6).

`r` is short for `rune`, and rune is a keyword. I'm open to suggestions here, but I wouldn't question this in code review, myself.